Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py

```diff
@@ -19,6 +25,111 @@ def makebuf(text):
     return BIO.MemoryBuffer(text)
 
 
+def create_pem_string(base64_cert) -> str:
+    """This function takes the base64 encoded certificate received via
+    ad-search in the 'UserCertificate' attribute and converts it to a PEM
+
+    Args:
+        base64_cert (_type_): _description_
+
+    Returns:
+        str: PEM string for the certificate
+    """
+    pemstring = '-----BEGIN CERTIFICATE-----\n'
+    pemstring += '\n'.join(
+        [base64_cert[i: i + 64] for i in range(0, len(base64_cert), 64)]
+    )
+    pemstring += '\n-----END CERTIFICATE-----\n'
+    return pemstring
+
+
+def parse_multipart_message(msg: str):
```

Comment:
This function is too long
I think we can divide it to 3 parts:
"handle_attachment", "extract_images_and_body", and "replace_cid_with_data_url".